model/product_model.py

```diff
@@ -0,0 +1,21 @@
+from typing import Optional
+from datetime import datetime 
+from pydantic import BaseModel, Field, model_validator
+
+class Product(BaseModel):
+        id: str
+        name: str
+        price: float = Field(gt=0, description="The price must be greater than zero")
+        quantity: int
+        active: bool = True
+        description: str
+        created_at: Optional[datetime] = None
+        updated_at: Optional[datetime] = None
+        @model_validator(mode='before')
```

Comment:
Vc n precisa fazer isso, da pra usar a anotation de valor default pro pydantic
https://docs.pydantic.dev/latest/concepts/fields/#default-values

Reply:
Perfeito. vou alterar e colocar a anotation de valor default.

Reply:
Ta quase la mas ainda não é isso. 
Na documentação, mais abaixo de onde ta marcado, mostra um outro tipo de default que é o default_factory, nele você pode passar uma função que vai retornar o valor que vc vai usar como default. Como por exemplo a função datetime.now().  E assim sempre que o objeto for criado, esse campo vai ser preenchido automaticamente com o horario atual.

Reply:
Entendi. com a classe factory eu consigo garantir que toda vez que a classe Product for instanciada e se o campo created_at não for passado o valor da data atual é gerado e com isso não preciso usar o @model_validator.